### PR TITLE
DOC: Cross Link full/full_like in a few see-also sections.

### PR DESCRIPTION
**Pull request by Carreau**
While teaching numpy I was asked the best way to create an array of `nan`,
and `np.full` seem not be cross linked from many places; In particular
in the documentation of `zeros` and `ones` seam like obvious candidates
to add them.

--- 

Here are two top answer on [Stack Overflow](https://stackoverflow.com/questions/1704823/initializing-numpy-matrix-to-something-other-than-zero-or-one) where user can can confused by [chaining `.fill(nan)`](https://stackoverflow.com/questions/22414152/best-way-to-initialize-and-fill-an-numpy-array) and get confused, and what the various options are.

### CODE REVIEW

File: numpy/add_newdocs.py [hunk 1]
```diff
@@ -923,6 +923,8 @@ def luf(lamdaexpr, *args, **kwargs):
     empty : Return a new uninitialized array.
     ones : Return a new array setting values to one.
     zeros : Return a new array setting values to zero.
+    full : Return a new array of given shape filled with value.
+    full_like : Return a new array with shape of input filled with value.
```
Comment (eric-wieser) [inline numpy/add_newdocs.py]: This would make more sense up under `zeros_like`

File: numpy/add_newdocs.py [hunk 2]
```diff
@@ -986,6 +988,8 @@ def luf(lamdaexpr, *args, **kwargs):
     zeros_like : Return an array of zeros with shape and type of input.
     ones_like : Return an array of ones with shape and type of input.
     empty_like : Return an empty array with shape and type of input.
+    full : Return a new array of given shape filled with given value.
+    full_like : Return a new array with shape of input filled with value.
```
Comment (eric-wieser) [inline numpy/add_newdocs.py]: Would be nice to keep the orders matching here too

**Review by eric-wieser — COMMENTED** [review]
I might be inclined to lengthen these links chains a little, and have:

* Each of `ones`, `zeros`, `empty`, and `full` link to each other
* Each of `ones_like`, `zeros_like`, `empty_like`, and `full_like` link to each other
* Link `ones` to `ones_like` and vice versa, `zeros` to `zeros_like`, etc.

Not compulsory though.

What I would like you to do before I merge is preserve the ordering and grouping of the function names

**Review by stefanv — COMMENTED** [review]
Thanks, Mattias, I ran into this just last week.

**Comment by Carreau** [comment]
Update the `array` See-Also, all descriptions across the 8-functions are the same, in same order (excluding link to self and link to non similar `_like` ness): 

    - empty_like
    - ones_like
    - zero_like
    - full_like
    - empty
    - ones
    - zeros
    - full

Thus each convenience function links to 4 other ones, and `array` links to the 8th of them.

**Comment by Carreau** [comment]
> Thanks, Matthias, I ran into this just last week.

That's thanks to @carvalhocamille that had well thoughts questions.

**Review by eric-wieser — APPROVED** [review]

**Comment by Carreau** [comment]
Thanks!

**Pull request merged**